pkg/modprovider/planstore.go

```diff
@@ -31,9 +31,41 @@ import (
 //
 // The provider will have a single planStore instance that stores the information indexed by URNs.
 type planStore struct {
-	mutex  sync.Mutex
-	plans  map[urn.URN]*planEntry
-	states map[urn.URN]*stateEntry
+	mutex     sync.Mutex
+	plans     map[urn.URN]*planEntry
+	states    map[urn.URN]*stateEntry
+	operation Operation
```

Comment:
There will be only one of these per process. That's by design right? Just confirming.

Reply:
Yes that's right. Right now it's used to track whether this process was triggered from a `moduleStateHandler.Delete` or by `Construct`.